targets/TARGET_STM/stm_spi_api.c

```diff
@@ -384,7 +396,108 @@ void spi_format(spi_t *obj, int bits, int mode, int slave)
     DEBUG_PRINTF("spi_format, bits:%d, mode:%d, slave?:%d\r\n", bits, mode, slave);
 
     // Save new values
-    handle->Init.DataSize          = (bits == 16) ? SPI_DATASIZE_16BIT : SPI_DATASIZE_8BIT;
+    uint32_t DataSize;
+    switch (bits) {
+#if defined(TARGET_STM32F0) || defined(TARGET_STM32F3) || defined(TARGET_STM32F7) || \
+    defined(TARGET_STM32G0) || defined(TARGET_STM32G4) || defined(TARGET_STM32L4) || \
+    defined(TARGET_STM32L5) || defined(TARGET_STM32WB) || defined(TARGET_STM32H7)
+        case 4:
+            DataSize = SPI_DATASIZE_4BIT;
+            break;
+        case 5:
+            DataSize = SPI_DATASIZE_5BIT;
```

Comment:
What do you think about code like:

````
    switch (bits) {
#if defined(SPI_DATASIZE_4BIT)
        case 4:
            DataSize = SPI_DATASIZE_4BIT;
            break;
#endif
#if defined(SPI_DATASIZE_4BIT)
        case 5:
            DataSize = SPI_DATASIZE_5BIT;
            break;
#endif
...
````

Reply:
It's a little more verbose on the `#if` flags, but I think it follows your general style better. Plus, it has the benefit of not having to port in new families when they hit, assuming that the semantics remain the same.

While looking at #13941 regarding the default output for async transfers, I noticed that there was at least one place where the count passed to the structure was transfers and not bytes. I will look for those locations and update them similarly.